flink-tests/src/test/java/org/apache/flink/test/checkpointing/EventTimeWindowCheckpointingITCase.java

```diff
@@ -299,12 +288,17 @@ public void apply(
 								sum += value.f1.value;
 								key = value.f0;
 							}
-							out.collect(new Tuple4<>(key, window.getStart(), window.getEnd(), new IntType(sum)));
+
+							final Tuple4<Long, Long, Long, IntType> result =
+								new Tuple4<>(key, window.getStart(), window.getEnd(), new IntType(sum));
+							out.collect(result);
 						}
 					})
-					.addSink(new ValidatingSink(numKeys, numElementsPerKey / windowSize)).setParallelism(1);
+				.addSink(new ValidatingSink<>(
+					new SinkValidatorUpdateFun(numElementsPerKey),
+					new SinkValidatorCheckFun(numKeys, numElementsPerKey, windowSize))).setParallelism(1);
 
-			tryExecute(env, "Tumbling Window Test");
+			env.execute("Tumbling Window Test");
```

Comment:
Why is this not using `tryExecute()` anymore?

Reply:
Because it does not require to use `SuccessExceptions` because in event time the end of the source function is deterministic.